docs/concepts/Security-Best-Practices.md

```diff
@@ -0,0 +1,166 @@
+
+-	[Visual Studio](https://visualstudio.microsoft.com/downloads/)
+-	[.NET SDK & Runtime](https://dotnet.microsoft.com/download)
+-	[NuGet](https://www.nuget.org/downloads)
+-	[NuGet packages](../consume-packages/reinstalling-and-updating-packages.md)
+
+## Manage your dependencies
+
+### NuGet deprecated and vulnerable dependencies
+
+You can use the [dotnet CLI](/dotnet/core/tools/dotnet-list-package) to list any known deprecated or vulnerable dependencies you may have inside your project or solution. You can use the command `dotnet list package --deprecated` or `dotnet list package --vulnerable` to provide you a list of any known deprecations or vulnerabilities.
+
+### GitHub vulnerable dependencies
+
+If your project is hosted on GitHub, you can leverage [GitHub Security](https://docs.github.com/en/free-pro-team@latest/github/finding-security-vulnerabilities-and-errors-in-your-code/automatically-scanning-your-code-for-vulnerabilities-and-errors) to find security vulnerabilities and errors in your project and Dependabot will fix them by opening up a pull request against your codebase. 
+
+Catching vulnerable dependencies before they are introduced is one goal of the [“Shift Left”](https://en.wikipedia.org/wiki/Shift-left_testing) movement. Being able to have information about your dependencies such as their license, transitive dependencies, and the age of dependencies helps you do just that.
+
+For more information about Dependabot alerts & security updates, [see the following documentation](https://docs.github.com/en/github/managing-security-vulnerabilities/about-alerts-for-vulnerable-dependencies).
+
+### NuGet feeds
+
+Packages can come from different feeds. To ensure you are secure, knowing what feed your packages are coming from is a best practice. One such best practice is the use of a single feed. You can accomplish this by using multiple upstream source feeds to bring your packages into a single feed.
+
+For more information to secure your package feeds, see [3 Ways to Mitigate Risk When Using Private Package Feeds](https://azure.microsoft.com/en-us/resources/3-ways-to-mitigate-risk-using-private-package-feeds/).
+
+### Client trust policies
+
+There are policies that you can opt-into in which you require the packages you use to be signed. This allows you to trust a package author, as long as it is author signed, or trust a package if it is owned by a specific user or account that is repository signed by NuGet.org.
+
+To configure client trust policies, [see the following documentation](../consume-packages/installing-signed-packages.md).
+
+### Lock files
+
+Lock files store the hash of your package’s content. If the content hash of a package you want to install matches with the lock file, it will ensure package repeatability.
+
+To enable lock files, [see the following documentation](../consume-packages/package-references-in-project-files#locking-dependencies).
+
+## Monitor your supply chain
```

Comment:
Those are all recommendations to publishers.. the title should be appropriate. Something like "Be a secure publisher".

Reply:
Maybe this belongs in a separate doc? I think of monitoring mostly as an author based activity. You monitor if people squat on your package id, steal your api secrets, try to log into your accounts, reserve your prefix, see a new vulnerability reported on your package, etc.

From a consumer standpoint there's less control you have because you don't do anything but managing your dependencies from what you "know".

I could put some type of indicators or something that says "Package Consumer" | "Package Author" which might help give this more clarity as the doc expands more advice.

Reply:
Agree. We need a clear separation between consumer recommendations and publishers. Both your suggestions - separate doc or just separate sections will work.

Reply:
Let me see what I can do. Would adding tags the sections under the title seem okay with the same structure given we will be adding as we learn more?

Reply:
For example:

## NuGet dependency graph
"Package Consumers" | "Package Authors"
Blah blah blah advice

Reply:
I don't have a strong opinion here :) tags LGTM